libs/hooks/debounce/src/lib/debounce-hook.ts

```diff
@@ -0,0 +1,190 @@
+import type {
+  EvaluationContext,
+  EvaluationDetails,
+  FlagValue,
+  Hook,
+  HookContext,
+  HookHints,
+} from '@openfeature/web-sdk';
+import { FixedSizeExpiringCache } from './utils/fixed-size-expiring-cache';
+
+/**
+ * An error cached from a previous hook invocation.
+ */
+export class CachedError extends Error {
+  private _innerError: unknown;
+
+  constructor(innerError: unknown) {
+    super();
+    Object.setPrototypeOf(this, CachedError.prototype);
+    this.name = 'CachedError';
+    this._innerError = innerError;
+  }
+
+  /**
+   * The original error.
+   */
+  get innerError() {
+    return this._innerError;
+  }
+}
+
+export type Options<T extends FlagValue = FlagValue> = {
+  /**
+   * Function to generate the cache key for the before stage of the wrapped hook.
+   * If the cache key is found in the cache, the hook stage will not run.
+   * If not defined, the DebounceHook will no-op for this stage (inner hook will always run for this stage).
```

Comment:
Why don't we default to using the 'flagKey'? It seems like that would be the most common use case. It looks like the hook doesn't do anything if you don't set the `beforeCacheKeySupplier`, but it's optional. I think we should make it either required or set a default. My preference would be to set a default.

Reply:
[This comment](https://github.com/open-feature/js-sdk-contrib/pull/1389#discussion_r2414225730) has implications here.